pkgs/tools/typesetting/tex/texlive/combine.nix

```
@@ -298,14 +291,17 @@ in (buildEnv {
 
   # MkIV uses its own lookup mechanism and we need to initialize
   # caches for it.
+  # We use faketime to improve determinism, but the uuid embedded
+  # in the cache is still not reproducible.
   ''
     if [[ -e "$out/bin/mtxrun" ]]; then
-      mtxrun --generate
+      FORCE_SOURCE_DATE=1 TZ= faketime -f '@1980-01-01 00:00:00 x0.001' mtxrun --generate
```

We could replace the uuids with a constant value with something like
```suggestion
      substitute $TEXMFDIST/scripts/context/lua/mtxrun.lua mtxrun.lua --replace "osuuid()" "'e2402e51-133d-4c73-a278-006ea4ed734f'"
      FORCE_SOURCE_DATE=1 TZ= faketime -f '@1980-01-01 00:00:00 x0.001' luatex --luaonly mtxrun.lua --generate
```
(uuid chosen by fair dice roll; guaranteed to be random). After some quick tests this appears to work fine. Note that the patched version of mtxrun.lua is not installed, just used for generating the initial cache.

---

Not generating the context filesystem tree at all doesn't work:
```
➜ ./result/bin/context a.tex 
mtxrun          | unknown script 'context.lua' or 'mtx-context.lua'
```

---

To be safe, we should use different uuid's for each call, so more like
```suggestion
      substitute "$TEXMFDIST"/scripts/context/lua/mtxrun.lua mtxrun.lua --replace 'cache_uuid=osuuid()' 'cache_uuid="e2402e51-133d-4c73-a278-006ea4ed734f"' --replace 'uuid=osuuid(),' 'uuid="242be807-d17e-4792-8e39-aa93326fc871",'
      FORCE_SOURCE_DATE=1 TZ= faketime -f '@1980-01-01 00:00:00 x0.001' mtxrun --generate
```
That seems reasonable.

Maybe we don't need `x0.001` here, it makes the output puzzling (it claims to have finished in `0.005` seconds). I'll check.

---

...and we need the `x0.001` as the timestamp seems to be the *end* time so it can change depending on the speed.

@apfelkuchen6 I would be grateful if you could double check that it all works as it seems to do on my machine!